docs/src/modules/spring/pages/views.adoc

```diff
@@ -1 +1,167 @@
+
+[#value-entity]
+== Creating a View from a Value Entity
+
+Consider an example of a Customer Registry service with a `customer` Value Entity. When `customer` state changes, the entire state is emitted as a value change. Value changes update any associated Views. To create a View that lists customers by their name:
+
+<<_define_the_view>> for a service that selects customers by name and associates a table name with the View. The table is created and used by Kalix to store the View.
+
+
+This example assumes the following `customer` exists
+
+[source,java,indent=0]
+.src/main/java/customer/api/Customer.java
+----
+include::example$spring-customer-registry-views-quickstart/src/main/java/customer/api/Customer.java[tags=class]
+----
+
+As well as a Value Entity service `CustomerEntity.java` that will produce the state changes consumed by the View.
+
+
+=== Define the View
+
+You implement a View by extending `kalix.javasdk.view.View` and subscribing to changes from an entity. You specify how to query it by providing a method annotated with `query`, which is then made accessible via REST annotations.
+
+.src/main/java/customer/view/CustomerByNameView.java
+[source,java]
+----
+include::example$spring-customer-registry-views-quickstart/src/main/java/customer/view/CustomerByNameView.java[tags=class]
+----
+<1> defining view ID
+<2> defining table name
+<3> subscribing to `CustomerEntity`
+<4> extending from `View`
+<5> defining endpoint
+<6> defining the query
+<7> note that no return is needed
+
+IMPORTANT: Note that the return value of the method is `null`. You may ask yourself,  how is it that the endpoint respond with any `Customer` at all?. When you call this endpoint, it first hits the proxy, which calls directly to the database. When the proxy receives the response, it sends it directly to you without any further intervention from the View. Therefore, null is valid as a return value in the endpoint of a View. The choice of `null` is our way to make clear that the response doesn't come from the return of this method. But you can choose any response you like as long is compatible with the return type. 
+
+Adding a view ID to your View allows you to refactor the name of the class later on without the risk of losing the history of events. If you don't define a view ID the class name becomes its ID. Therefore, if you change the name of your class afterwards Kalix will not recognize this new name as the same view and will create a brand new view. This is resource consuming for a view from an entity because it will reprocess all the event of that entity to rebuild the view. While for a view built from a topic you will loose all the previous events because these views only process events from current time forwards.
+
```

Comment:
It's worth adding an example showing how to transform from `Customer` to `CustomerView`. 

We can add on this PR or later on. If on follow-up, let's create an issue so we don't forget.

Reply:
https://github.com/lightbend/kalix/issues/7884